benchmarks/_benchmark.py

```diff
@@ -237,24 +253,33 @@ def _handle_error(
         self,
         e: Exception,
         name: str,
+        options: Dict[str, Any],
         tags: Dict[str, Any],
         info: Dict[str, Any],
         context: Dict[str, Any],
         r_command: Optional[str] = None,
     ) -> Tuple[Dict[str, Any], None]:
         output = None
         tags["name"] = name
-        error = {
+        result = {
+            "run_name": options.get("run_name")
+            or f"{options.get('run_reason')}: {self.github_info['commit']}",
+            "run_id": self.conbench.get_run_id(options=options),
+            "run_reason": options.get("run_reason"),
+            "batch_id": options.get("batch_id") or self.conbench._batch_id,
             "timestamp": _now_formatted(),
             "tags": tags,
             "info": info,
             "context": context,
-            "error": str(e),
+            "error": {"error": str(e), "stack_trace": traceback.format_exc()},
+            "optional_benchmark_info": {},
+            "machine_info": self.conbench.machine_info,
+            "github": self.github_info,
         }
         if r_command is not None:
-            error["command"] = r_command
-        logging.exception(json.dumps(error))
-        return error, output
+            result["error"]["command"] = r_command
+        logging.exception(f"Errored result (not posted): {json.dumps(result)}")
```

Comment:
Theoretically we _could_ post these to Conbench now, but I'm not sure whether there's value in doing so. Errors within the benchmark won't hit this—this is for when things bomb out entirely somehow, e.g. you spelled the name of the benchmark wrong.